lib/facter/self_service.rb

```diff
@@ -87,4 +87,21 @@
     postgresversion = PuppetSelfService.pe_postgres_service_name
     { S0011: PuppetSelfService.service_running_enabled(postgresversion.to_s) }
   end
+
+  chunk(:S0012) do
+    next unless PuppetSelfService.replica? || PuppetSelfService.postgres? || PuppetSelfService.primary? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler?
+    # Did Puppet Produce a report in the last run interval
+    summary_path = Puppet.settings['lastrunfile']
+    lastrunfile = YAML.load_file(summary_path)
+    time_lastrun = lastrunfile['time']['last_run']
```

Comment:
A malformed summary file could cause `time_lastrun = lastrunfile['time']['last_run']` to fail if `lastrunfile['time']` is `nil`. Maybe use `time_lastrun = dig('time', 'last_run')` which will return `nil` if it is not found. You could then return false or do a `next` if `time_lastrun.nil?`